src/cargo/ops/lockfile.rs

```diff
@@ -19,20 +20,25 @@ pub fn load_pkg_lockfile(pkg: &Package) -> CargoResult<Option<Resolve>> {
 
 pub fn load_lockfile(path: &Path, sid: &SourceId) -> CargoResult<Option<Resolve>> {
     // If there is no lockfile, return none.
-    let mut f = match File::open(path) {
-        Ok(f) => f,
+    let s = match load_string(path) {
```

Comment:
D'oh. I kept thinking: this can't be right, why is the error just ignored. Turns out I misread the code.
Would you be interested in using `paths::read` here and ignoring `ErrorKind::NotFound` specifically but propagating anything else?